macOS/debugger.cpp

```diff
@@ -239,18 +305,44 @@ void Debugger::GetLoadCommand(mach_header_64 mach_header,
   }
 }
 
+void *Debugger::MakeSharedMemory(mach_vm_address_t address, size_t size) {
+  mach_port_t shm_port;
+  if (address == 0)
+    return NULL;
+
+  memory_object_size_t memoryObjectSize = round_page(size);
+  kern_return_t ret = mach_make_memory_entry_64(mach_target->Task(), &memoryObjectSize, address, VM_PROT_READ, &shm_port, MACH_PORT_NULL);
+  if (ret != KERN_SUCCESS) {
+    FATAL("Error (%s) remote allocate share memory\n", mach_error_string(ret));
+  }
+
+  mach_vm_address_t map_address;
```

Comment:
Should be initialized to NULL, otherwise mach_vm_map might fail despite VM_FLAGS_ANYWHERE